commit 27cef1144071f38f9807a239ca0f3e8ea7e554a3
Author: root <root@dhcp-flab2-52.cesnet.cz>
Date:   Wed Jun 21 11:02:15 2017 +0200

    allow dash in client names

diff --git a/warden3/files/opt/warden_server/warden_server.py b/warden3/files/opt/warden_server/warden_server.py
index ba4d95d..0c1e55c 100755
--- a/warden3/files/opt/warden_server/warden_server.py
+++ b/warden3/files/opt/warden_server/warden_server.py
@@ -1441,7 +1441,7 @@ def modify_client(**kwargs):
             for label in hostname.split("."))
 
     def isValidNSID(nsid):
-        allowed = re.compile("^(?:[a-zA-Z_][a-zA-Z0-9_]*\\.)*[a-zA-Z_][a-zA-Z0-9_]*$")
+        allowed = re.compile("^(?:[a-zA-Z_][a-zA-Z0-9_\-]*\\.)*[a-zA-Z_][a-zA-Z0-9_\-]*$")
         return allowed.match(nsid)
 
     def isValidEmail(mail):
