commit 07f0452f5951360170a566c59385320ee22bb42c
Author: root <root@dhcp-flab2-51.cesnet.cz>
Date:   Thu Jun 22 10:02:15 2017 +0200

    return code on specific event (double registering client)

diff --git a/warden3/files/opt/warden_server/warden_server.py b/warden3/files/opt/warden_server/warden_server.py
index d4a694b..76b3a3a 100755
--- a/warden3/files/opt/warden_server/warden_server.py
+++ b/warden3/files/opt/warden_server/warden_server.py
@@ -1478,7 +1478,7 @@ def modify_client(**kwargs):
     for c in server.handler.db.get_clients():
         if kwargs["name"] is not None and kwargs["name"].lower()==c.name:
             print >>sys.stderr, "Clash with existing name: %s" % str(c)
-            return 254
+            return 101
         if kwargs["secret"] is not None and kwargs["secret"]==c.secret:
             print >>sys.stderr, "Clash with existing secret: %s" % str(c)
             return 254
